tests/pass-dep/shims/fcntl.rs

```diff
@@ -0,0 +1,14 @@
+//@ignore-target-windows: no libc on Windows
+//@compile-flags: -Zmiri-isolation-error=warn-nobacktrace
+
+use std::io::Error;
+
+fn main() {
+    // test `fcntl`
+    // This should succeed for everything.
+    unsafe {
+        assert_eq!(libc::fcntl(1, libc::F_DUPFD, 0), 3);
+        // Since there's no error, this might not have the right behavior.
+        assert_eq!(Error::last_os_error().raw_os_error(), Some(0));
+    }
+}
```

Comment:
Why a new test? Seems fine to keep this in libc-fs-with-isolation, but assert that it returns 0 and no error?

Also why do you check last_os_error when there was no error...?

Reply:
I wasn’t sure the right route to go for this test case. I moved it to a different file since it didn’t seem like it fit with the other functions erroring in libc-fs-with-isolation. I moved it back.

How would I assert there’s no error without checking last_os_error? The function doesn’t return 0 because it is returning a duplicated file descriptor.